.github/workflows/relay.yml

```
@@ -0,0 +1,80 @@
+name: Relay
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "relay/**"
+      - ".github/workflows/relay.yml"
+  pull_request:
+    branches:
+    paths:
+      - "relay/**"
+      - ".github/workflows/relay.yml"
+  workflow_dispatch:
+
+jobs:
+  build-test-publish:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./relay
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-relay-cache
+          key: ${{ runner.os }}-buildx-relay-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-relay-
+
+      - run: sudo df -h
+
+      - name: Build Docker Build Image
+        uses: docker/build-push-action@v2
+        with:
+          context: ./relay
+          target: gobuild
+          tags: go-build-env
+          load: true # load image to local registry to use it in next steps
+          cache-from: type=local,src=/tmp/.buildx-relay-cache
+          cache-to: type=local,dest=/tmp/.buildx-relay-cache
+
+      - name: Run Go tests
+        run: |
+          docker run \
+            --workdir /go/src/github.com/keep-network/tbtc/relay \
+            go-build-env \
+            gotestsum
+
+      - name: Login to Google Container Registry
+        if:  |
+          startsWith(github.ref, 'refs/heads/releases/')
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.GCR_REGISTRY_URL }}
+          username: _json_key
+          password: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
+
+      - name: Build and publish Docker Runtime Image
+        uses: docker/build-push-action@v2
+        env:
+          IMAGE_NAME: 'relay'
+          GOOGLE_PROJECT_ID: ${{ secrets.KEEP_TEST_GOOGLE_PROJECT_ID }}
+        with:
+          context: ./relay
+          # GCR image should be named according to following convention:
+          # HOSTNAME/PROJECT-ID/IMAGE:TAG
+          # We don't use TAG yet.
+          tags: ${{ secrets.GCR_REGISTRY_URL }}/${{ env.GOOGLE_PROJECT_ID }}/${{ env.IMAGE_NAME }}
+          labels: revision=${{ github.sha }}
+          build-args: |
+            REVISION=${{ github.sha }}
+          push: |
+            ${{ startsWith(github.ref, 'refs/heads/releases/') }}
```

Same thing as with the comment for line 58:
```suggestion
          push: |
            ${{ github.ref == 'refs/heads/master'
              && (github.event_name == 'push'
              || github.event_name == 'workflow_dispatch') }}
```

---

Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0